frontend/database.go

```diff
@@ -2,38 +2,127 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"os"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
 )
 
-var (
-	databaseURL  string = os.Getenv("DB_URL")
-	databaseName string = os.Getenv("DB_NAME")
+const (
+	clustersContainer = "Clusters"
 )
 
+// DBClient defines the needed values to perform CRUD operations against the async DB
+type DBClient struct {
+	client *azcosmos.Client
+	DBName string
+	DBUrl  string
+}
+
+// NewDatabaseClient instanstiates a Cosmos DatabaseClient targeting Frontends async DB
+func NewDatabaseClient() *DBClient {
+	cred, _ := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
No strong feelings here, I imagine it would be easy to move out but im not sure i understand the benefit and am curious.

My original thinking was when a new frontend is created, it creates the DBClient as part of it which requires the creds so I made it all part of the new client setup

```
	f := &Frontend{
		logger:   logger,
		listener: listener,
		metrics:  emitter,
		server: http.Server{
			ErrorLog: slog.NewLogLogger(logger.Handler(), slog.LevelError),
			BaseContext: func(net.Listener) context.Context {
				return ContextWithLogger(context.Background(), logger)
			},
		},
		cache:    *NewCache(),
		dbClient: *NewDatabaseClient(),
```

as always, open to learning better ways to do it haha

Reply:
Ahh i follow what you mean now. With the workload identity setup in place, this call is basically using injected environment variables that are setup with the workload identity. I could always add the options object and leave it unset and pass it so its essentially passing nothing but gives us options later to add to it? Based on some previous cards and DDR's i think workload identity is definite so the type of credential will remain the same as you mentioned but doesnt mean we won't need to tweak things. Possibly a card to dig into the options would be beneficial before implementing to see if there is something we want for sure now?